File: edukate-backend/src/main/java/io/github/sanyavertolet/edukate/backend/permissions/SubmissionPermissionEvaluator.java
package io.github.sanyavertolet.edukate.backend.permissions;

import io.github.sanyavertolet.edukate.backend.entities.Submission;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@AllArgsConstructor
public class SubmissionPermissionEvaluator {
    public Boolean isOwner(Submission submission, String userId) {
        return submission.getUserId().equals(userId);
    }
}

File: edukate-backend/src/main/kotlin/io/github/sanyavertolet/edukate/backend/permissions/SubmissionPermissionEvaluator.kt
package io.github.sanyavertolet.edukate.backend.permissions

import io.github.sanyavertolet.edukate.backend.entities.Submission
import org.springframework.stereotype.Component

@Component
class SubmissionPermissionEvaluator {
    fun isOwner(submission: Submission, userId: String): Boolean = submission.userId == userId
}
